File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.Arrays;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        Arrays.sort(people); // 정렬
        int a = 0;
        int b = people.length - 1;
        while(a <= b) {
            if(people[a] + people[b] > limit) { // 가장 무거운 사람과 가장 가벼운 사람이 같이 못타면 가장 무거운 사람 혼자 보내기
                b--;
                answer++;
            }
            else {
                a++;
                b--;
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort()
    answer = 0
    a = 0
    b = len(people) - 1
    
    while a <= b:
        if people[a] + people[b] > limit:
            b -= 1
        else:
            a += 1
            b -= 1
        answer += 1
        
    return answer